File: CSE3063F22_iteration2_classes/Advisor.java
import java.io.FileNotFoundException;
import java.io.IOException;

import org.json.simple.parser.ParseException;

public class Advisor {
    
    public void ApproveRegistration(Student std) throws FileNotFoundException, IOException, ParseException{
        std.setStatus(getRandomBoolean(70));
    };

    public boolean getRandomBoolean(float probability) {
        double randomValue = Math.random()*100;  //0.0 to 99.9
        return randomValue <= probability;
    };
}

File: python_code/Advisor.py
import random
from Student import Student
class Advisor(Person,object):

    def approve_registration(self, std:Student):
        std.set_status (self.get_random_boolean(70))
    
    def get_random_boolean(self, probability):
        random_value = random.random() * 100  # 0.0 to 99.9
        return random_value <= probability
